### DISCUSSION TITLE
New DB screws up metadata

### DISCUSSION THREAD

**Discussion by Michel-Giesecke** [RafaelGB/obsidian-db-folder#374] [Q&A] [answered]
I wanted to create a database with the plugin for my movie database. I used the tag I had previously set for movies and managed to create the page. However, upon restarting obsidian the metadata of the movie files is screwed up. New lines are added and the YAML is invalid. Consequently, the db cannot find the files anymore and I have to manually fix every files metadata.

Anybody got an idea why this is happening?

**Comment by RafaelGB** [comment]
Hi @Michel-Giesecke , v2.5.0 brings a lot of changes and (in theory) the bugs you are describing were fixed today. Which version are you using?

**Reply by Michel-Giesecke** [reply]
I checked and I have the current version (2.5.3) of the plugin installed

**Comment by RafaelGB** [comment]
@Michel-Giesecke  I moved the issue to the discussion because you select an empty issue template

**Comment by Michel-Giesecke** [comment]
I just set up a test and figured something out. I think the issue arises when renaming columns. When I rename columns it screws up the YAML frontmatter by adding invalid lines.

**Reply by Michel-Giesecke** [reply]
I think the issue is with how changing headers for columns changes the value in the actual file. This behavior is not clear on first glance and apparently also has some bugs if you try to change the values that are not defined in the actual frontmatter, but later on with `year:: 2000` for example

**Reply by RafaelGB** [reply]
If you change the label, modify all the metadata of your notes, yes.

Maybe something like a duplicate field or a yaml breaker character its the reason. Could you share some console log?

**Comment by Michel-Giesecke** [comment]
I tried changing an inline datafields name and it again added invalid YAML lines and threw the following error.

```
app.js:1 
Error: [CM6] tree.iterate has changed syntax.
See the stack trace to find the faulty plugin and file an issue with the plugin author. 
Details: https://discuss.codemirror.net/t/release-0-20-0/4302
    at app.js:1:1541194
    at e.iterate (app.js:1:293053)
    at ei.iterate (app.js:1:1541428)
    at LivePlugin.build (plugin:obsidian-react-components:145519:59)
    at LivePlugin.update (plugin:obsidian-react-components:145480:26)
    at e.update (app.js:1:412381)
    at e.updatePlugins (app.js:1:498015)
    at e.update (app.js:1:496247)
    at e._dispatch (app.js:1:493593)
    at e.dispatch (app.js:1:495249)
```

Edit: I quickly checked and even with metaedit disabled the same error is thrown. So there is no connection there

**Reply by Michel-Giesecke** [reply]
The frontmatter should look like this:

```
---
created: 2022-05-21
type: movie
imdbId:
 - tt1981677
length: 1 hours, 52 minutes
tags: db/movietest
---
```

But after the bug it always ends up looking like this:
```
---
created: 2022-05-21
type: movie
imdbId:
 undefined:
  - tt1981677
 - tt1981677
length: 1 hours, 52 minutes
tags: db/movietest
---
```

**Comment by RafaelGB** [comment, marked as answer]
is your imdbId column of tag type? if not it can not edit a list properly

The log looks like an error with https://github.com/elias-sundqvist/obsidian-react-components plugin

**Reply by Michel-Giesecke** [reply]
I now tried it with both metaedit and react components disabled. If I now change the column name it manages to change the name of the infield value, but it still messes up the YAML frontmatter. However the console does not output any errors. So I guess the issue really is with the list character of the imdbId value

**Reply by Michel-Giesecke** [reply]
I think I found the issue. The value imdbId was used as both a frontmatter value and an inline vaule in my template. If I remove either the frontmatter is not invalid anymore

**Reply by Michel-Giesecke** [reply]
Thank you for your help figuring out my issue. Everything is working as intended now. Sorry for the user error

**Reply by RafaelGB** [reply]
Nice! We can improve the msg feedback with those cases
